src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDEFailureInfoBarTests.cs

```diff
@@ -189,6 +187,33 @@ public async Task InfoBarButtonClicked_OutputWindowIsShown()
             outputWindowService.VerifyNoOtherCalls();
         }
 
+        [TestMethod]
+        public async Task ShowAsync_VerifySwitchesToUiThreadIsCalled()
+        {
+            var infoBarManager = new Mock<IInfoBarManager>();
+            infoBarManager.Setup(x => x.AttachInfoBarWithButton(ValidToolWindowId, It.IsAny<string>(), It.IsAny<string>(), default));
+
+            var threadHandling = new Mock<IThreadHandling>();
+
+            var testSubject = new OpenInIDEFailureInfoBar(infoBarManager.Object, Mock.Of<IOutputWindowService>(), threadHandling.Object);
+
+            await testSubject.ShowAsync(ValidToolWindowId);
+
+            threadHandling.Verify(x => x.RunOnUIThread(It.IsAny<Action>()), Times.Once);
```

Comment:
I can't easily test if the methods that get called are running in the ui thread as they are private so I decided to just test if RunOnUIThread gets called.

Reply:
How about testing ` infoBarManager.AttachInfoBarWithButton` ?